src/app/games/hive/tests/HiveRules.spec.ts

```diff
@@ -0,0 +1,961 @@
+
+            // When moving the beetle by one space
+            const move: HiveMove = HiveMove.move(new Coord(0, 0), new Coord(0, 1)).get();
+
+            // Then the move should succeed
+            const expectedBoard: Table<HivePiece[]> = [
+                [[], [Q], [q], [b]],
+                [[B], [], [], []],
+            ];
+            const expectedState: HiveState = HiveState.fromRepresentation(expectedBoard, 5);
+
+            RulesUtils.expectMoveSuccess(rules, state, move, expectedState);
+        });
+        it('should forbid moving beetle by more than one space', () => {
+            // Given a state with the player's beetle on the board
+            const board: Table<HivePiece[]> = [
+                [[B], [Q], [q], [b]],
+            ];
+            const state: HiveState = HiveState.fromRepresentation(board, 4);
+
+            // When moving the beetle by two space
+            const move: HiveMove = HiveMove.move(new Coord(0, 0), new Coord(1, 1)).get();
+
+            // Then the move should fail
+            const reason: string = HiveFailure.BEETLE_CAN_ONLY_MOVE_TO_DIRECT_NEIGHBORS();
+            RulesUtils.expectMoveFailure(rules, state, move, reason);
+        });
+        it('should allow moving beetle on top of another piece', () => {
+            // Given a state with the player's beetle on the board
+            const board: Table<HivePiece[]> = [
+                [[B], [Q], [q], [b]],
+            ];
+            const state: HiveState = HiveState.fromRepresentation(board, 4);
+
+            // When moving the beetle on top of its neighbor
+            const move: HiveMove = HiveMove.move(new Coord(0, 0), new Coord(1, 0)).get();
+
+            // Then the move should succeed
+            const expectedBoard: Table<HivePiece[]> = [
+                [[], [B, Q], [q], [b]],
+            ];
+            const expectedState: HiveState = HiveState.fromRepresentation(expectedBoard, 5);
+
+            RulesUtils.expectMoveSuccess(rules, state, move, expectedState);
+        });
+        it('should allow moving the beetle when it is on top of another piece', () => {
+            // Given a state with the player's beetle on top of another piece
+            const board: Table<HivePiece[]> = [
+                [[B, Q], [q], [b]],
+            ];
+            const state: HiveState = HiveState.fromRepresentation(board, 4);
+
+            // When moving the beetle to an adjacent space
+            const move: HiveMove = HiveMove.move(new Coord(0, 0), new Coord(1, 0)).get();
+
+            // Then the move should succeed
+            const expectedBoard: Table<HivePiece[]> = [
+                [[Q], [B, q], [b]],
+            ];
+            const expectedState: HiveState = HiveState.fromRepresentation(expectedBoard, 5);
+
+            RulesUtils.expectMoveSuccess(rules, state, move, expectedState);
+        });
+        it('should support up to the 4 beetles on top of another, with another piece underneath', () => {
```

Comment:
should support having the four beetle on top of a piece ?

Reply:
ja